test/e2e/tests/basic_test.go

```diff
@@ -0,0 +1,122 @@
+// +build e2e
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/yahoo/athenz/clients/go/zms"
+	"github.com/yahoo/k8s-athenz-syncer/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	domain         = zms.DomainName("k8s.omega.stage.kube-test")
+	domainStr      = "k8s.omega.stage.kube-test"
+	trustdomain    = zms.DomainName("prod-eng.omega.acceptancetest.trust-domain")
+	trustdomainStr = "prod-eng.omega.acceptancetest.trust-domain"
+	namespace      = "prod--eng-omega-acceptancetest-test--domain"
+	nsdomain       = "prod-eng.omega.acceptancetest.test-domain"
+	roleName       = zms.EntityName("syncer-e2e")
+	roleNameStr    = zms.ResourceName(domainStr + ":role." + "syncer-e2e")
+	roleResource   = zms.ResourceName("syncer-e2e")
+	role           = zms.Role{
+		Name: roleResource,
+	}
+	trustroleName     = zms.EntityName("test-trustrole")
+	trustroleResource = zms.ResourceName("test-trustrole")
+	trustRole         = zms.Role{
+		Name:  trustroleResource,
+		Trust: trustdomain,
+	}
+)
+
+func TestBasicRoleUpdate(t *testing.T) {
+	f := framework.Global
+	err := f.ZMSClient.PutRole(domain, roleName, "", &role)
+	if err != nil {
+		t.Errorf("Unable to add role")
+	}
+	time.Sleep(90 * time.Second)
```

Comment:
We can use `wait.PollImmediate` to wait for the cr to be updated instead of hard coding the 90 second timeout. Example: https://github.com/yahoo/k8s-athenz-istio-auth/blob/master/test/integration/authz_v1_test.go#L49-L62